src/modules/fancyzones/tests/UnitTests/ZoneWindow.Spec.cpp

```diff
@@ -120,88 +140,89 @@ namespace FancyZonesUnitTests
         {
-            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, m_monitor, m_deviceId.c_str(), m_virtualDesktopId.c_str(), false);
+            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, m_monitor, m_uniqueId.str(), false);
             testZoneWindow(m_zoneWindow);
             Assert::IsNull(m_zoneWindow->ActiveZoneSet());
         }
 
         TEST_METHOD(CreateZoneWindowNoZoneWindowHost)
         {
-            m_zoneWindow = MakeZoneWindow(nullptr, m_hInst, m_monitor, m_deviceId.c_str(), m_virtualDesktopId.c_str(), false);
+            m_zoneWindow = MakeZoneWindow(nullptr, m_hInst, m_monitor, m_uniqueId.str(), false);
 
             Assert::IsNull(m_zoneWindow.get());
         }
 
         TEST_METHOD(CreateZoneWindowNoZoneWindowHostFlashZones)
         {
-            m_zoneWindow = MakeZoneWindow(nullptr, m_hInst, m_monitor, m_deviceId.c_str(), m_virtualDesktopId.c_str(), true);
+            m_zoneWindow = MakeZoneWindow(nullptr, m_hInst, m_monitor, m_uniqueId.str(), true);
 
             Assert::IsNull(m_zoneWindow.get());
         }
 
         TEST_METHOD(CreateZoneWindowNoHinst)
         {
-            m_zoneWindow = MakeZoneWindow(m_hostPtr, {}, m_monitor, m_deviceId.c_str(), m_virtualDesktopId.c_str(), false);
+            m_zoneWindow = MakeZoneWindow(m_hostPtr, {}, m_monitor, m_uniqueId.str(), false);
 
             testZoneWindow(m_zoneWindow);
             Assert::IsNull(m_zoneWindow->ActiveZoneSet());
         }
 
         TEST_METHOD(CreateZoneWindowNoHinstFlashZones)
         {
-            m_zoneWindow = MakeZoneWindow(m_hostPtr, {}, m_monitor, m_deviceId.c_str(), m_virtualDesktopId.c_str(), true);
+            m_zoneWindow = MakeZoneWindow(m_hostPtr, {}, m_monitor, m_uniqueId.str(), true);
 
             testZoneWindow(m_zoneWindow);
             Assert::IsNull(m_zoneWindow->ActiveZoneSet());
         }
 
         TEST_METHOD(CreateZoneWindowNoMonitor)
         {
-            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, {}, m_deviceId.c_str(), m_virtualDesktopId.c_str(), false);
+            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, {}, m_uniqueId.str(), false);
 
             Assert::IsNull(m_zoneWindow.get());
             Assert::IsNotNull(m_hostPtr);
         }
 
         TEST_METHOD(CreateZoneWindowNoMonitorFlashZones)
         {
-            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, {}, m_deviceId.c_str(), m_virtualDesktopId.c_str(), true);
+            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, {}, m_uniqueId.str(), true);
 
             Assert::IsNull(m_zoneWindow.get());
             Assert::IsNotNull(m_hostPtr);
         }
 
         TEST_METHOD(CreateZoneWindowNoDeviceId)
         {
-            m_zoneWindow = MakeZoneWindow(m_hostPtr, m_hInst, m_monitor, nullptr, m_virtualDesktopId.c_str(), false);
+            // Generate unique id without device id
+            std::wstring uniqueId = GenerateUniqueId(m_monitor, {}, m_virtualDesktopId.c_str());
```

Comment:
Change `deviceId` argument to `nullptr` to be consistent with following test (`CreateZoneWindowNoDesktopId`).